internal/installutil/adoption.go

```diff
@@ -0,0 +1,69 @@
+package installutil
+
+import (
+	"context"
+	"errors"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"time"
+)
+
+func AdoptRunner(ctx context.Context, ui terminal.UI, client pb.WaypointClient, id string, addr string) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Waiting for runner to connect to server at %s...", addr)
+	defer func() { s.Abort() }()
+
+	// Waits 5 minutes for the server to detect the new runner before timing out
+	d := time.Now().Add(time.Minute * time.Duration(5))
+	ctx, cancel := context.WithDeadline(ctx, d)
+	defer cancel()
```

Comment:
nit: could probably just use `time.NewTimer()` here: https://pkg.go.dev/time#NewTimer

Reply:
@catsby upon updating this to the below using a timer, I get stuck in the loop 🤔 

```suggestion
	// Waits 5 minutes for the server to detect the new runner before timing out
	timer := time.NewTimer(time.Minute * time.Duration(5))
	defer timer.Stop()
	found := false
	for !found {
		select {
		case <-timer.C:
		case <-ctx.Done():
			ui.Output("Cancelled.",
				terminal.WithErrorStyle(),
			)
			return errors.New("context canceled")
		}
```